### ISSUE TITLE
 Dropdowns  > Locations > Items  empty when selecting Cameras Items

### ISSUE THREAD

**Issue by Firewolf1337** [glpi-project/glpi#16561]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.12

### Bug description

In the Locations dropdown setting there is a section for Items which is basically showing a dropdown menu to select a Item type.( see screenshot)
If I select "Cameras Items" the dropdown will disappear and nothing else is shown. It is Location independent and even adding a Camera is not resolving the issue.

![image](https://github.com/glpi-project/glpi/assets/18141890/3967b147-af4e-4101-a0f4-38e44ec57874)

### Relevant log output

```shell
php-errors.log

[2024-02-15 07:53:52] glpiphplog.CRITICAL:   *** Uncaught Exception RuntimeException: Cannot build an empty union query in /usr/share/nginx/html/glpi/src/QueryUnion.php at line 101
  Backtrace :
  src/DBmysqlIterator.php:285                        QueryUnion->getQuery()
  src/DBmysqlIterator.php:111                        DBmysqlIterator->buildQuery()
  src/DBmysql.php:1109                               DBmysqlIterator->execute()
  src/Location.php:459                               DBmysql->request()
  src/Location.php:395                               Location->showItems()
  src/CommonGLPI.php:694                             Location::displayTabContentForItem()
  ajax/common.tabs.php:120                           CommonGLPI::displayStandardTab()
```

### Steps To reproduce

1. Go to:  Setup > Dropdowns > Locations
2. Select any Location
3. Select "Items" tab
4. Select "Cameras items"

To redo it clear als cookies.

**Comment by trasher** [comment]
I confirm the issue

**Comment by trasher** [comment]
I'm working on a PR to get that fixed; but this requires a database change (among others). A quick fix would be to remove this faulty type from the configuration:

```diff
diff --git a/inc/define.php b/inc/define.php
index d7de4131c6..8e8e550c9e 100644
--- a/inc/define.php
+++ b/inc/define.php
@@ -333,7 +333,10 @@ $CFG_GLPI["socket_types"]                  = ['Computer','NetworkEquipment',
 
 $CFG_GLPI['itemdevices'] = [];
 foreach ($CFG_GLPI['device_types'] as $dtype) {
-    $CFG_GLPI['location_types'][] = 'Item_' . $dtype;
+    //quick'n'dirty fix for #16561
+    if ($dtype !== 'DeviceCamera') {
+        $CFG_GLPI['location_types'][] = 'Item_' . $dtype;
+    }
     $CFG_GLPI["itemdevices"][] = 'Item_' . $dtype;
 }
```

**Comment by Firewolf1337** [comment]
Awesome, thanks for investigating and solving.
For me the dirty way works since we don't have any Cameras.

Thanks for the "fix"!
